src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2274,9 +2274,20 @@ static void updateAttrInNotifyCer
       //
       // FIXME P6: https://github.com/telefonicaid/fiware-orion/issues/2587
       // If an attribute has no value, then its value is not updated (neither is previousValue).
-      // However this may be problematic ... see the issue
+      // However this may be problematic ... see the issue.
       //
+      // New data on this: the functest
+      // "test/functionalTest/cases/2998*/null_not_working_in_q_for_subscription.test" fails with this 'if-clause' (not outdeffed),
+      // and removing the 'if' the functest 'only q_and_mq_as_uri_param_for_metadata.test' fails, but it seems like the test is
```

Comment:
cases/xxxx*/q_and_mq_as_uri_param_for_metadata.test also ;)

Reply:
Fixed in 9b203a5a6